### PR TITLE
Submission Progress Check - Tribal Data Release

### PR DESCRIPTION
**Pull request by jperson1**
# Submission Progress Check - Tribal Data Release

This PR goes into the feature branch [jppd/tribal-consent-form](https://github.com/GSA-TTS/FAC/pull/2238), not main. 

Issue: https://github.com/GSA-TTS/FAC/issues/2102
See conversation here: https://github.com/GSA-TTS/FAC/pull/2238#discussion_r1340627099

## Changes:
1. In the view:
    * Check to see if the current user is the auditee certifier, pass in context
    * Check to see if tribal data release form is complete, pass in context
2. In the checklist HTML template:
    * Using the new view variables, five display cases exist for tribal data consent
        1. Does not appear at all for non-tribal entities.
        2. The auditee certifying official sees the link and can complete it.
        3. The form is complete. The auditee certifying official can edit it.
        4. Anyone else with access sees an inactive section, no link.
        5. The form is complete. Anyone else with access sees that it's done but still can't click a link.

## How to test:
1. Switch to this branch and run normally
2. Make some new audits.
    * One where the org is _not_ tribal
    * One where the org _is_ tribal, and you're the auditee certifier
    * One where the org _is_ tribal, but you're _not_ the auditee certifier
3. Run through those audits to the checklist. Very that things look correct, according to the five display cases.
4. Complete the tribal data consent form. Verify that things look correct.

If you want to edit the tribal data release object, do so through the admin panel rather than something like DBeaver. 

## Screenshots:
1. Does not appear at all for non-tribal entities.
image

2. The auditee certifying official sees the link and can complete it.
image

3. The form is complete. The auditee certifying official can edit it.
image

4. Anyone else with access sees an inactive section, no link.
image

5. The form is complete. Anyone else with access sees that it's done but still can't click a link.
image

## PR checklist: submitters

- [x]   Link to an issue if possible. If there’s no issue, describe what your branch does. Even if there is an issue, a brief description in the PR is still useful.
- [x]   List any special steps reviewers have to follow to test the PR. For example, adding a local environment variable, creating a local test file, etc.
- [ ]   For extra credit, submit a screen recording like [this one](https://github.com/GSA-TTS/FAC/pull/1821).
- [x]   Make sure you’ve merged `main` into your branch shortly before creating the PR. (You should also be merging `main` into your branch regularly during development.)
- [ ]   Make sure that whatever feature you’re adding has tests that cover the feature. This includes test coverage to make sure that the previous workflow still works, if applicable.
- [x]   Do manual testing locally. Our tests are not good enough yet to allow us to skip this step. If that’s not applicable for some reason, check this box.
- [x]   Verify that no Git surgery was necessary, or, if it was necessary at any point, repeat the testing after it’s finished.
- [ ]   Once a PR is merged, keep an eye on it until it’s deployed to dev, and do enough testing on dev to verify that it deployed successfully, the feature works as expected, and the happy path for the broad feature area (such as submission) still works.

## PR checklist: reviewers

- [ ]   Pull the branch to your local environment and run `make docker-clean; make docker-first-run && docker compose up`; then run `docker compose exec web /bin/bash -c "python manage.py test"`
- [ ]   Manually test out the changes locally, or check this box to verify that it wasn’t applicable in this case.
- [ ]   Check that the PR has appropriate tests. Look out for changes in HTML/JS/JSON Schema logic that may need to be captured in Python tests even though the logic isn’t in Python.
- [ ]   Verify that no Git surgery is necessary at any point (such as during a merge party), or, if it was, repeat the testing after it’s finished.
        
The larger the PR, the stricter we should be about these points.

### CODE REVIEW

File: backend/audit/templates/audit/submission_checklist/submission-checklist.html [hunk 1]
```diff
@@ -61,16 +61,61 @@
...
+                            {% else %}
+                            {% comment %} 3. The form is complete. The auditee certifying official can edit it. {% endcomment %}
+                                {% if is_user_auditee_certifier %}
+                                    {% include './icon-list-icon.html' with completed=True %}
+                                    <div class="text-success-darker">
+                                        <p class="text-bold text-underline margin-0">Tribal data release (Complete)</p>
+                                        <p class="margin-0">
+                                            Auditees must opt in or opt out of making their reporting package publicly available.
+                                        </p>
+                                    </div>
```
Comment (danswick) [inline backend/audit/templates/audit/submission_checklist/submission-checklist.html:105]:
```suggestion
                                    <div class="text-success-darker">
                                        <p class="text-bold text-underline margin-0">Tribal data release (Complete)</p>
                                        <p class="margin-0">
                                            Auditees must opt in or opt out of making their reporting package publicly available.
                                        </p>
                                        <p class="margin-0">
                                            <a class="usa-link"
                                                href="{% url 'audit:TribalAuditConsent' report_id %}">Edit the tribal data release form</a>
                                        </p>
                                    </div>
```

I wasn't able to edit while testing. Should we add an edit link like this?

**Review by danswick — CHANGES_REQUESTED** [review]
This is looking good! While testing the `certifying auditee, form complete` case locally, I wasn't able to go back and edit again. I added a suggestion inline, but not sure if that's the ideal way to handle an edit link outside of the usual "sections" way of doing things.

**Comment by eazapata23** [comment]
> This is looking good! While testing the `certifying auditee, form complete` case locally, I wasn't able to go back and edit again. I added a suggestion inline, but not sure if that's the ideal way to handle an edit link outside of the usual "sections" way of doing things.

I ran into the same scenario `certifying auditee, form complete`, I like what you suggested. This will help with the cypress tests as they currently only have a happy path.

File: backend/audit/templates/audit/submission_checklist/submission-checklist.html (hunk 1 above)
Comment (jperson1) [reply backend/audit/templates/audit/submission_checklist/submission-checklist.html]: And this is why we have reviews. Thanks for the catch! Fixed in code and updated the screenshots.

**Comment by jperson1** [comment]
The above issue has been fixed - missed an edit link. Also, I moved the link up above the cross validation link.

**Pull request merged**